src/compressed_tensors/quantization/lifecycle/initialize.py

```diff
@@ -89,13 +100,33 @@ def _initialize_scale_zero_point_observer(
 
     device = next(module.parameters()).device
 
+    # infer expected scale/zero point shape
+    expected_shape = 1  # per tensor
+    if (
+        not hasattr(module, "quantization_status")
+        or getattr(module, "quantization_status") == QuantizationStatus.FROZEN
+    ):
+        expected_shape = 0
+
+    if base_name == "weight" and weight_shape is not None:
+        if quantization_args.strategy == QuantizationStrategy.CHANNEL:
+            # (output_channels, 1)
+            expected_shape = (weight_shape[0], 1)
+        elif quantization_args.strategy == QuantizationStrategy.GROUP:
+            expected_shape = (
+                weight_shape[0],
+                weight_shape[1] // quantization_args.group_size,
+            )
+
     # initializes empty scale and zero point parameters for the module
     init_scale = Parameter(
-        torch.empty(0, dtype=torch.float16, device=device), requires_grad=False
+        torch.empty(expected_shape, dtype=torch.float16, device=device),
```

Comment:
Does the expected shape matter here? Won't it just get overridden when we update the scale and zero_point?

Reply:
on the compressed tensors side - it could. for HFQuantizer, we need the shapes to match otherwise it will run into a state dict load error